partiql-test-data/fail/semantic/primitives/call.ion

```diff
@@ -0,0 +1,111 @@
+trim::[
+    semantic::{
```

Comment:
Nit.: Just occurred to me and might be applicable to other files as well—could all these tests be under one `semantic` namespace?

Reply:
Could you clarify "under one `semantic` namespace? Is the suggestion to wrap all the tests in a top-level namespace called `semantic`? Such as

```
semantic::[
    trim::[
        semantic::{
	        name: "TRIM too many arguments",
	        statement: "TRIM(BOTH ' ' FROM 'test' 2)",
	        assert: {
	            result: SemanticError
	        },
	    },
	    ... // other trim tests
	]
	extract::[
		... // extract tests
	]
	... // other tests
]
```

The most recent commit, https://github.com/partiql/partiql-tests/pull/11/commits/26a5720d8eba843e9f4d8421d5c25f511eefb449, added the `semantic` test annotation (different from `parse` or `eval`). I'm not too sure what having a `semantic` namespace provides.

Reply:
Sorry, perhaps I should've been more clear; I think the ambiguity is because I'm still trying to wrap my head around the test model. The motivation for my comment was to have the tests in such a way that we can avoid repeating the annotation (in this case `semantic`). This is not a blocker and I know requires more changes in case it makes sense. So feel free to merge and we can discuss further later.

Reply:
Ah, I get what you mean. The test model is still subject to change as we continue defining tests, especially as we define the formal schema and verification in #3. Regarding the test category annotation, I'd say the primary benefits are that
- the parsing of the test cases by the test runner doesn't need to keep track of the test category from an outer namespace or file path
  - e.g. `partiql-lang-rust` parsing of test case: https://github.com/partiql/partiql-lang-rust/blob/5acb46609fc5e9e57494d77b137c394fe7c60622/partiql-conformance-test-generator/src/lib.rs#L111-L141
- could allow for mixing different test categories within a file/namespace

Reply:
Yep I totally get it, the model is to be finalized. On the same note, I don't see the point of tracking the test category when we have the Assertion. Basically, if there is a `ParseError`, I think it already gives enough information about the failure.

Reply:
Yeah there seems to be some unnecessary redundancy involving the assertion's `result` and the category annotation. The annotation only provides the test category, while the assertion `result` provides the test category **and** the expected result (e.g. ok or error). Some further discussion will be needed to see if this test category annotation is necessary.